rfcs/text/onChange.md

```diff
@@ -0,0 +1,131 @@
+- Start Date: 2019-10-23
+
+# Summary
+
+`onChange` & `onBlur` interface standardisation
+
+# Basic example
+
+```js
+const internalChange = e => {
+    e.target = {
+      ...(name && { name }),
+      ...(id && { id }),
+      value: [
+        {
+          displayText: e.target.innerText,
+          optionValue: e.target.value
+        }
+      ]
+    };
+
+    onChange(e);
+  };
+```
+
+# Motivation
+
+Currently the `onChange` and `onBlur` handlers in Carbon use a mix of custom objects `{value: "my-value"}` and React's `SyntheticEvent`.
+
+In the custom objects, `name` or `id` are passed inconsistently.
+
+The purpose of this RFC is to standardise the approach, so users can rely upon the interface between Carbon and their application.
+
+The approach that we choose should be consistent across components so users familiar with the interface of one component can easily use another component.
+
+# Detailed design
+
+Users who implement this should be familiar with React's [SyntheticEvent](https://reactjs.org/docs/events.html).
+
+When writing a Carbon component the user should wrap the user provided `onChange`/`onBlur` callback with code similar to that in
+the basic example.
+
+* Components that provide a formatted value **MUST** replace `SyntheticEvent.target`.
+* `SyntheticEvent.target` **MUST** be an object with the properties `name`, `id`, `value`.
+* `SyntheticEvent.target` **MUST** include the `name` and `id` of the component (if the component has these props).
+```jsx
+<Select id="foo" name="foo">
```

Comment:
It would be great, to have values that are more in line with how the component could be used. 

foo, bar, baz has caused us so many problems when visiting a file written by someone else and you want to use it or make changes. 

Using something like, `name='Visible in the component'` and `value='id of visible value'`, make it easier to follow or maybe supplying context with, "when I have a set of data with, `{ key1: 'value1', key2: 'value2' }` and showing where those values would be used."

Reply:
I've addressed this comment by giving a more realistic example in 4b01825.